util/util.go

```diff
@@ -38,10 +38,17 @@ func SdkConfigPath() (string, error) {
 	return filepath.Join(homeDir, ".config", "gcloud"), nil
 }
 
+// unixHomeDir returns the user's home directory.  Note that $HOME has
+// precedence over records in the password database since the credential helper
+// may be running under a different UID in a user namespace.
 func unixHomeDir() string {
+	homeDir := os.Getenv("HOME")
+	if homeDir != "" {
+		return homeDir
+	}
 	usr, err := user.Current()
 	if err == nil {
-		return usr.HomeDir
+		homeDir = usr.HomeDir
```

Comment:
Stylistic nit, I think I'd prefer something like this just to lexically scope things as tightly as we can:

```go
if home := os.Getenv("HOME"); home != "" {
  return home
}

if usr, err := user.Current(); err == nil {
  return usr.HomeDir
}

return ""
```

Reply:
Sounds good, thanks!